src/lib/components/common/ui/tooltip.ts

```diff
@@ -0,0 +1,47 @@
+import {css, html} from 'lit';
+import {FloatElement} from '../../custom';
+import {CustomElement} from '../../injectors';
+import {property} from 'lit/decorators.js';
+
+import {hintcss} from './hintcss';
+
+@CustomElement()
+export class Tooltip extends FloatElement {
+    @property({type: String}) class = '';
+    @property({type: String}) text!: string;
+
+    static get styles() {
+        return [
+            ...FloatElement.styles,
+            hintcss,
+            css`
+                [class*='hint--'][aria-label]:after {
+                    text-shadow: none;
+                    font-family: 'Motiva Sans', Arial, Helvetica, sans-serif;
+                    font-weight: normal;
+                    line-height: normal;
+                    text-align: center;
+                    background: #c2c2c2;
+                    color: #3d3d3f;
+                    font-size: 11px;
+                    border-radius: 3px;
+                    padding: 5px;
+                }
+                .hint--whitespace-pre-wrap:after,
+                .hint--whitespace-pre-wrap:before {
+                    white-space: pre-wrap;
+                }
+            `,
+        ];
+    }
+
+    render() {
+        const tooltipClass = `hint--top hint--rounded hint--no-arrow ${this.class}`;
+
+        return html`
+            <div class="${tooltipClass}" aria-label="${this.text}" style="display: block;">
```

Comment:
It seems like we could avoid forcing the caller to a somewhat _rigid_ DOM structure that requires a `block` element if we move to using a Lit Directive (https://lit.dev/docs/templates/custom-directives/).

This would potentially change it to something like:

```
<div class="market-float-bar-container" style="left: ${left}%; width: ${dynamicWidth.toFixed(2)}%;">
    ${tooltip("hey this is a tooltip")}
    ...
</div>
```

You'd need to include `hintcss` in the base FloatElement styles I believe. I don't have a strong preference of using a directive however, so if it proves to be annoying then we can ditch it -- just figured we should try to reduce the DOM complexity.

Reply:
Sorry for the late reply! I absolutely do see your point here. As this is my first project with Lit, I wasn't quite familiar with Directives yet.

I switched to a Lit Directive for the tooltip now in 4eae18c. It indeed does look a bit cleaner, especially in regards to the DOM structure.

So the workflow with the new implementation is now:
- import the directive and use it as `${tooltip("this is a tooltip", "optional-extraclass")}`
- import `tooltipStyles` and add it in the component-styles with `static styles = [...tooltipStyles]`, which contains both `hintcss` as well as the custom tooltip styling

Do let me know if this matches your initial idea with Directives.

Reply:
Thanks @GODrums ! Overall, it looks good!

I think you can just put the tooltip styles by default in `FloatElement` and can rename `tooltipDirective` to just `tooltip`.